include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -2972,46 +2794,37 @@ namespace internal
 
 
 
-  template <int dim, typename VectorizedArrayType>
+  template <int dim, typename Number>
```

Comment:
I agree to some extent, but then we get back to the main problem: This code is designed to be agnostic to the type, so essentially al code in `evaluation_kernels.h` should be possible to instantiate with a generic `Number` (say `double`) rather than `VectorizedArray`. I think the right choice would be to define the `Number` template of `FEEvaluationBaseData` as `VectorizedArrayType` (and in turn also `MappingInfoStorage`), as those classes actually hold all data in SIMD type.
Since we're breaking things now anyway, we might as well simplify things. The question is if we need to maintain backward compatibility here. `FEEvaluationBaseData` was introduced about a year ago with https://github.com/dealii/dealii/commit/09aa10961063fc8f12ed35542e96aa39ec533d21#diff-ccfe43c7e151d41aeafc2b1091a20db30a424d7b41b4d4841eeb438f7e5feb06 so we might need to add a short alias. What do you think?

Reply:
> essentially al code in `evaluation_kernels.h` should be possible to instantiate with a generic `Number` (say `double`) rather than `VectorizedArray`

OK, this holds for all code except the `gather_evaluate` and `integrate_scatter` functions, of course. But I would still lean towards that simplification.